### PR TITLE
enable global settings in parsing topology.

### LINKED ISSUE #2578: How to globally set some parameters related to network optimization?
Previously, we have some functions to globally set some parameters. In v2 API, the way to globally set these parameters changes a lot.

Can someone help me to make sure how each of them changes? Becuase these parameters are crucial to learning performance.

1. parameters moving into `Optimizer` (**but they cannot be correctly recognized, a terrible BUG** https://github.com/PaddlePaddle/Paddle/issues/2488)
    ```python

    default_momentum(val)
    default_decay_rate(val)
    default_gradient_clipping_threshold(val)
    ```
2. parameters I am not sure how to set.
    - It seems that the default initialization strategy becomes smart initialization,
    - but if I do not want it, for example, just use simple uniform initialization, how can I do to change the initialization method (and then only modify some parameters by hand), instead of modifying each parameter one by one.

    ```python
    default_initial_smart(val)
    default_initial_std(val)
    default_initial_mean(val)
    default_initial_strategy(val)
     ```
 3. parameters I not sure that they are still supported.
    - for training use both GPU and CPU ``` default_device(val) ``` 
        - some layer in PaddlePaddle does not support running in GPU, for example, CRF layer. But running the entire model on CPU is really slow, in this situation, I hope to run the computation extensive parts of the model on GPU while the rest on CPU.
        - Previously, I can set a default device id, for example, GPU as default, and only change the device id of CRF layer to make it run on CPU, I am not sure how to run a model that some parts on GPU while other parts on GPU.

    - for model compression
    ```python
    default_update_hooks(val)
    default_compact_func(val)
    ```
    - for cluster training
    ```python
    default_num_batches_regularization(val)
    ```

### LINKED ISSUE #3006: Implementation of paddle.attr.Param ignore the globally set initialization mean and std, and force me to use smart initalization
There is a code snippet from `paddle/trainer_config_helpers/attrs.py` https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/trainer_config_helpers/attrs.py#L139:

```python
    def __init__(self,
                 name=None,
                 is_static=False,
                 initial_std=None,
                 initial_mean=None,
                 initial_max=None,
                 initial_min=None,
                 l1_rate=None,
                 l2_rate=None,
                 learning_rate=None,
                 momentum=None,
                 gradient_clipping_threshold=None,
                 sparse_update=False,
                 update_hooks=None,
                 initializer=None):
        self.attr = {}

        if is_static:
            self.attr['is_static'] = True

        if initial_std is None and initial_mean is None and initial_max \
                is None and initial_min is None:
            self.attr['initial_smart'] = True
```
* in training, I may prefer another initialization strategy,  such as initialization from a Gaussian distribution with a default mean and std, rather than use smart initialization.
* to do this, I will set the global initialization std and mean.
* The implementation above ignores the globally set mean and std. If I just use `paddle.attr.Param` to tune the learning rate or any other parameters except `initial_std`, `initial_mean`, `initial_max`, and `initial_min` like below. The codes then force me to use smart initialization. It ignores the default initialization std I set.

    ```python
    param_attr=paddle.attr.Param(learning_rate=0.1)
    ```

### PR DESCRIPTION
**Pull request by lcy-seso**
1. fixes https://github.com/PaddlePaddle/Paddle/issues/3006
2. fixes https://github.com/PaddlePaddle/Paddle/issues/2578

### CODE REVIEW

File: python/paddle/trainer_config_helpers/attrs.py [hunk 1]
```diff
@@ -58,23 +58,31 @@ def is_compatible_with(x, Type):
...
+
+    Reference:
+        Learning both Weights and Connections for Efficient Neural Networks
+        https://arxiv.org/pdf/1506.02626.pdf
+
+    The example usage is:
 
-    :param sparsity_ratio: Must be specified if hook type is 'pruning', 
-                        it represents the ratio of the zero elements to be set by the Parameter.
+    .. code-block:: python
```
Comment (luotao1) [inline python/paddle/trainer_config_helpers/attrs.py]: 71-72行中空一行，不然代码显示不了。

**Review by luotao1 — APPROVED** [review]
LGTM

**Comment by luotao1** [comment]
感谢您给PaddlePaddle贡献代码。由于Paddle V1/V2版本已不再维护，相关代码也已从develop分支上删除，因此关闭您的PR，欢迎您向Paddle最新版-Fluid贡献代码。
Thanks for contributing to PaddlePaddle! Since V1/V2 will not be maintained anymore, and related codes have been deleted from develop branch as well, we close this PR. Welcome to contribute to Fluid——the latest version of PaddlePaddle.
